mtags/src/main/scala-3/scala/meta/internal/pc/completions/Completions.scala

```diff
@@ -102,33 +145,64 @@ class Completions(
   inline private def undoBacktick(label: String): String =
     label.stripPrefix("`").stripSuffix("`")
 
-  private def findSuffix(methodSymbol: Symbol): Option[String] =
-    if shouldAddSnippet && methodSymbol.is(Flags.Method) then
-      lazy val extensionParam = methodSymbol.extensionParam
-
-      val paramss =
-        if methodSymbol.is(Flags.Extension) then
-          methodSymbol.paramSymss.filter(
-            !_.contains(extensionParam)
-          )
-        else methodSymbol.paramSymss
-
-      paramss match
-        case Nil => None
-        case List(Nil) => Some("()")
-        case _ if config.isCompletionSnippetsEnabled =>
-          val onlyParameterless = paramss.forall(_.isEmpty)
-          lazy val onlyImplicitOrTypeParams = paramss.forall(
-            _.exists { sym =>
-              sym.isType || sym.is(Implicit) || sym.is(Given)
-            }
-          )
-          if onlyParameterless then Some("()" * paramss.length)
-          else if onlyImplicitOrTypeParams then None
-          else Some("($0)")
-        case _ => None
-      end match
-    else None
+  private def getParams(symbol: Symbol) =
+    lazy val extensionParam = symbol.extensionParam
+    if symbol.is(Flags.Extension) then
+      symbol.paramSymss.filter(
+        !_.contains(extensionParam)
+      )
+    else symbol.paramSymss
+
+  private def findSuffix(symbol: Symbol): Option[String] =
+    val bracesSuffix =
+      if shouldAddSnippet && symbol.is(
+          Flags.Method
+        ) && !isTypePosition
+      then
+        val paramss = getParams(symbol)
+        paramss match
+          case Nil => ""
+          case List(Nil) => "()"
+          case _ if config.isCompletionSnippetsEnabled =>
+            val onlyParameterless = paramss.forall(_.isEmpty)
+            lazy val onlyImplicitOrTypeParams = paramss.forall(
+              _.exists { sym =>
+                sym.isType || sym.is(Implicit) || sym.is(Given)
+              }
+            )
+            if onlyParameterless then "()" * paramss.length
+            else if onlyImplicitOrTypeParams then ""
+            else "($0)"
+          case _ => ""
+        end match
+      else ""
+
+    val bracketSuffix =
```

Comment:
interestingly, the `$0` is only inserted in the square bracket in such cases.

Reply:
I think it's still needed. **Currently** we never put multiple suffixes with `$0`  at the same time.

However, it's extremely difficult for me to understand there won't be multiple `$0` because each of them is not separated by `if/else if/else`.  We (Metals developers in the future) have to read all of the code around here to check all those conditions won't be all true in any cases, to prevent Metals from putting multiple braces. That leads to the code being error-prone.